openhands/storage/conversation/file_conversation_store.py

```diff
@@ -95,6 +98,48 @@ async def search(
         next_page_id = offset_to_page_id(end, end < num_conversations)
         return ConversationMetadataResultSet(conversations, next_page_id)
 
+    async def search_by_filters(
```

Comment:
Actually i think we should implement this method outside of the file conversation store

There are also other store implementations that exist. Maybe see if we can reuse the existing [`search_conversation`](https://github.com/All-Hands-AI/OpenHands/blob/main/openhands/server/routes/manage_conversations.py#L220) endpoint but with the added constraints you're looking for